### DISCUSSION TITLE
navigationDestination(isPresented:destination:)

### DISCUSSION THREAD

**Discussion by tospery** [pointfreeco/swift-composable-architecture#3421] [Q&A]
### Description

How to resolve warning in 04-Navigation-Lists-NavigateAndLoad.swift file:
struct NavigateAndLoadListView: View {
    @Bindable var store: StoreOf<NavigateAndLoadList>
    
    var body: some View {
        Form {
            Section {
                AboutView(readMe: readMe)
            }
            ForEach(store.rows) { row in
                NavigationLink(
                    "Load optional counter that starts from \(row.count)",
                    tag: row.id,
                    selection: .init(
                        get: { store.selection?.id },
                        set: { id in store.send(.setNavigation(selection: id)) }
                    )
                ) {
                    if let store = store.scope(state: \.selection?.value, action: \.counter) {
                        CounterView(store: store)
                    } else {
                        ProgressView()
                    }
                }
            }
        }
        .navigationTitle("Navigate and load")
    }
}
the warning is:
'init(_:tag:selection:destination:)' was deprecated in iOS 16.0: use NavigationLink(value:label:), or navigationDestination(isPresented:destination:), inside a NavigationStack or NavigationSplitView

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

**Comment by stephencelis** [comment]
That case study is using some older SwiftUI APIs which are deprecated, and are there just as examples for folks still targeting <iOS 16. There are other more modern examples, as well (see "Navigation Stacks"), but we can't show them both off in the same target and avoid the warnings.

Because this isn't a library bug, and so I'm going to convert it to a discussion, but if you have specific questions we can help with, let us know!

**Comment by tospery** [comment]
Since I am for ios 16, I want to resolve this warning, so I suggest you provide an example for ios 16 and above, thank you.

**Reply by VilemKurz** [reply]
Yes yes yes 🙏 . I am having a hard time to find out how to use NavigationLink _with tree based navigation_ - is that even possible on iOS 16+ without resorting to deprecated methods? For deep hierarchies it is unwieldy (and error prone) to pass delegation methods between childs through the root scene. Another drawback - imagine two screens from the stack are used in other stacks - then their delegation wiring must be duplicated in each stack root. Does anyone have an example, how to use NavigationLink with tree based navigation?

**Reply by stephencelis** [reply]
@VilemKurz SwiftUI doesn't really support this style of navigation, unfortunately. While there are a couple tree-based APIs via view modifier (`navigationDestination(isPresented/item)`), they are not well-supported compared to stack-based navigation. To get the best, least buggy experience with SwiftUI navigation we strongly recommend you adopt its stack-based APIs.
